### PR TITLE
provide Bool attribute to check for parents

### LINKED ISSUE #140: How to process dijkstra shortest path parents that return #undef?
This may be a more general Julia question, but I'm at a loss as to how to treat something like this:
```
h = graph(["a","b","c","d"],[])
add_edge!(h,"a","b"); add_edge!(h,"b","c"); add_edge!(h,"a","c"); add_edge!(h,"c","d")


julia> dijkstra_shortest_paths(h,"b").parents
4-element Array{ASCIIString,1}:
 #undef
    "b"
    "b"
    "c"
```
How do I test for this `#undef` in code?
Also, more of an enhancement request, I guess - is there ANY way to get Graphs to return something consistent when a node is not reachable? That is, strings return `#undef`, ints return either some very large number or `0` (not sure why):

```
julia> dijkstra_shortest_paths(g,2).parents
4-element Array{Int64,1}:
 140253310593744
               2
               2
               3
...

julia> dijkstra_shortest_paths(g,2).parents
4-element Array{Int64,1}:
 0
 2
 2
 3

``` 
and I'm not quite sure how to test for all these cases other than checking to see whether the value is in `graph.vertices`, which apparently fails with `ERROR: access to undefined reference` if the value is `#undef`...

### PR DESCRIPTION
**Pull request by yeesian**
closes #140.

### CODE REVIEW

File: src/dijkstra_spath.jl [hunk 1]
```diff
@@ -250,34 +256,25 @@ dijkstra_shortest_paths{V}(
...
-                currpind = vertex_index(currparent, g)
+    hasparent = state.hasparent
+    
+    nstates = length(parents)
+    all_paths = Array(Vector{V},nstates)
+    for i in 1:nstates
+        all_paths[i] = V[]
+        child_index = i
+        if hasparent[child_index]
+            println(child_index)
```
Comment (mlubin) [inline src/dijkstra_spath.jl]: debug printing?

**Comment by yeesian** [comment]
I can continue to work on support for `enumerate_paths` (in https://github.com/JuliaLang/Graphs.jl/pull/141), and change the title accordingly, before we merge this PR

**Comment by mlubin** [comment]
Smaller PRs are preferable. Please update the docs for ``dijkstra_shortest_paths`` and we can merge.

**Comment by yeesian** [comment]
Okay, I've updated the docs.

File: doc/source/algorithms.rst [hunk 2]
```diff
@@ -163,6 +163,7 @@ The following is an example that shows how to use this function:
 The result has several fields, among which the following are most useful:
 
 * ``parents[i]``:  the parent vertex of the i-th vertex. The parent of each source vertex is itself.
+* ``hasparent[i]``:  ``true`` if the i-th vertex has a parent, and ``false`` otherwise. Note that ``hasparent[i] == true`` for all source vertices.
```
Comment (mlubin) [inline doc/source/algorithms.rst]: I would add some interpretation for the case where ``hasparent[i] == false``. I.e., this means that the vertex isn't reachable from any source.

**Comment by yeesian** [comment]
Should I attempt squashing the commits again? xP

**Comment by mlubin** [comment]
That would be helpful. I don't want to get in trouble for merging messy PRs :)

**Comment by yeesian** [comment]
I think I got it right this time?

**Comment by mlubin** [comment]
Nope, somehow old commits have been included and the PR isn't mergeable.

**Comment by yeesian** [comment]
Yikes, I think I did a `git pull` by accident

**Comment by yeesian** [comment]
Is there a way to remedy it? Or should I do a new PR? haha

**Comment by mlubin** [comment]
Use rebase to delete the bad commits?

**Comment by yeesian** [comment]
Haha, I think I should just start afresh. I wonder what went wrong this time though.. did I rebase more commits than I had to? My understanding was that `pick` would leave the older logs untouched, and collapse only those that I `squash` into the latest `pick`.

**Pull request closed without merging**